File: src/task_1/observer.cpp
#include <geometry_msgs/PointStamped.h>
#include <ros/ros.h>

void poseCallback(const geometry_msgs::PointStamped &msg) {

  ROS_INFO("Point coords are: x:%f y:%f z:%f\n", msg.point.x, msg.point.y,
           msg.point.z);
}

int main(int argc, char **argv) {

  ros::init(argc, argv, "observer");
  ros::NodeHandle n;

  ros::Subscriber sub = n.subscribe("runner_position", 1000, poseCallback);

  ros::spin();

  return 0;
}

File: src/task_1/observer.py
#!/usr/bin/env python
import rospy
from geometry_msgs.msg import Point

def callback(position):
    rospy.loginfo("I heard x: %f, y: %f, z: %f", position.x, position.y, position.z)

def observer():

    rospy.init_node('observer', anonymous=True)
    rospy.Subscriber("runner_position", Point, callback)

    rospy.spin()

if __name__ == '__main__':
    observer()
